saleor/tests/fixtures.py

```diff
@@ -1061,6 +1061,9 @@ def product_with_images(product_type, category, media_root):
         category=category,
         is_published=True,
     )
+    ProductVariant.objects.create(
```

Comment:
Why? 

`product_type` have `size_attribute` as `variant_attribute`. 
If you want to create a Variant for this product you should assign value for this attribute.